File: backend/src/bond_monitor/infrastructure/paths.py
"""Shared filesystem paths for infrastructure layer."""

from __future__ import annotations

import os
from pathlib import Path

_REPO_ROOT = Path(__file__).resolve().parents[4]


def get_cache_dir() -> Path:
    """Return cache directory (overridable via CACHE_DIR env)."""
    return Path(os.getenv("CACHE_DIR") or (_REPO_ROOT / "cache"))


def get_ratings_json_path() -> Path:
    """Return path to vendored ratings.json."""
    return _REPO_ROOT / "data" / "ratings.json"

File: backend/internal/infrastructure/paths/paths.go
package paths

import (
	"os"
	"path/filepath"
	"runtime"
)

func repoRoot() string {
	if root := os.Getenv("BOND_MONITOR_ROOT"); root != "" {
		return root
	}
	_, file, _, ok := runtime.Caller(0)
	if !ok {
		return "."
	}
	// backend/internal/infrastructure/paths -> repo root
	return filepath.Clean(filepath.Join(filepath.Dir(file), "..", "..", "..", ".."))
}

// CacheDir returns cache directory (overridable via CACHE_DIR env).
func CacheDir() string {
	if dir := os.Getenv("CACHE_DIR"); dir != "" {
		return dir
	}
	return filepath.Join(repoRoot(), "cache")
}

// RatingsJSONPath returns path to vendored ratings.json.
func RatingsJSONPath() string {
	return filepath.Join(repoRoot(), "data", "ratings.json")
}
